### ISSUE TITLE
Qiskit module for Quantum Circuit Born Machines

### ISSUE THREAD

**Issue by jorgeplazas** [qiskit-advocate/qamp-2025#64]
# QAMP Project Proposal: Qiskit module for Quantum Circuit Born Machines

## Description

Quantum Circuit Born Machines are generative models that use quantum circuits to represent probability distributions. The expressive power of these models makes them a good candidate for tackling difficult unsupervised generative modeling tasks, specially those where a classic approach might fall short. Also, because of their relatively simple architecture, Quantum Circuit Born Machines are a great testbed for nearterm devices (cf. https://journals.aps.org/pra/abstract/10.1103/PhysRevA.98.062324, https://www.nature.com/articles/s41534-020-00288-9, and references therein).

This project aims at developing a set of tools for implementing Quantum Circuit Born Machines in such a way that these can be used to learn the distribution of a given dataset and sample new high quality instances of the learned distribution. 

## Deliverables

**Primary Deliverables:**
- A qiskit module for the implementation of Quantum Circuit Born Machines.
- A jupyter notebook serving both as a tutorial on QCBMs and as an example for the use of the module.
- Ideally if time permits a report on example performance metrics and comparisons with classical counterparts as benchmarks.

**Minimal Viable Product (MVP):**
- A set jupyter notebooks fully implementing use cases of Quantum Circuit Born Machines within the qiskit patterns framework. The code should implement functions that sample instances of the learned distributions. These notebooks should be documented in such a way as to be reusable in applications and provide a good learning aid for this class of models. 

### I am a MENTEE seeking a mentor:
**Type of mentor sought:** It would be great to work with someone with experience in the qiskit ecosystem and the implementation of modules and APIs within it. 

**Desired mentoring style:** Regular weekly check-ins and code reviews would be great. 

### As a MENTEE proposing this project:
**Name:** Jorge Plazas  
**GitHub:** [@jorgeplazas]  
**What I do:** I am a mathematical physicist, I work as an assistant professor a Universidad Francisco de Vitoria in Madrid. Currently I am working in projects that involve hybrid classical-quantum machine learning models. 

**Learning goals:** Quantum Circuit Born Machines are an essential component of an ongoing project in which I am the team lead. I am interested in implementing these in a framework which is general enough as to be able to explore the capabilities of these models and take full advantage of current hardware and software tools. 

**Additional Information:**
- **Time commitment:** ~ 6 to 8 hours a week.

**Comment by ashishar** [comment]
Hi,

I am keen to contribute to this project as a mentee to deepen my foundations in quantum computing/QML. I am pursuing PhD (Computer Science & Engineering) in QML from IIIT Sonepat, India. My profile on the Google scholar is below:
https://scholar.google.com/citations?user=AQ7zxvsAAAAJ&hl=en

Thank you so much for considering me.

Discord: _ashisharya

@ashishar

**Comment by Astri-Cornish** [comment]
Hello! @nrhawkins has been assigned as mentor to this project - thank you!

**Comment by Debshata** [comment]
Hello 👋,

I’m **Debshata Choudhury**, a dual-degree student (**B.S. Data Science – IIT Madras** & **B.Tech CSE – Amity University Noida**) and an **IBM Qiskit Advocate ’25**. I’m truly fascinated about Qiskit module for Quantum Circuit Born Machines methods—and I’d love to be part of this journey to push its open-source impact further.

### 🔹 Some of my work:
- **QuantAir** – Quantum-enhanced air-pollution forecasting (**QBoost over XGBoost**) → [Repo](https://github.com/V4I8H4VJ4IN62/QuantAir)  
- **VaahanBandhu** – Circular-logistics optimization with quantum-aided pathfinding → [Repo](https://github.com/Niranjan1Praveen/VahaanBandhu)  
- **Research:** *Comparative Study on Quantum Computing Qubits* → [Post](https://lnkd.in/p/gm2xbDUx) • [Paper](https://zenodo.org/records/15072694)

### 🎓 Credentials:
- [Qiskit Global Summer School 2024 – Advanced Badge](https://www.credly.com/badges/3381781d-d6b6-4630-8de7-b63a7c129aef/public_url)  
- [IBM Quantum Challenge 2024](https://lnkd.in/p/gNN8fHjs)  
- [QBronze Diploma](https://lnkd.in/p/gvx539ZA)

I’m deeply curious about building **practical, interpretable quantum models** and would be thrilled to collaborate, learn, and contribute to this project.

📫 **Discord:** debshatachoudhury | **LinkedIn:** Debshata Choudhury

**Comment by jorgeplazas** [comment]
# Qiskit QCBMs  
# Checkpoint 2 report
 
Debshata Choudhury,  Natalie Hawkins, and Jorge Plazas


We describe our progress in **Qiskit QCBMs**, a project where we develop a set of tools for the implementation of Quantum Circuit Born Machines within the Qiskit 2.x and Qiskit Patterns framework. This project is part of Qiskit Advocate Mentorship Program QAMP 2025.  

	
## The project in a Nutshell

Quantum Circuit Born Machines (QCBMs) are generative models in which the probability distribution of a dataset is represented in terms of a quantum state via Born's rule. QCBMs are known for their expressive power and have been widely recognized as a useful component in hybrid algorithms with a potential for quantum advantage. 


Our aim is to take advantage of various tools in the Qiskit ecosystem in order to optimize QCBMs for their use in near term devices.  Qiskit QCBMs is being developed as a Qiskit add-on which will be accompanied by a set of notebooks both showcasing the tools through examples and serving as an introduction to the theory. 

## Workflow, progress and status of the project

Our team, consisting of two mentees together with one mentor, has held weekly meetings since the beginning of the program in October 2025. 

At this point we have:

- Built a [github repository](https://github.com/jorgeplazas/Qiskit-QCBMs) where the code developed so far is hosted. The repo also contains a substantial amount of information on QCBMs and documents the evolution of the project.

-Reviewed the state of the art both in the literature and in avaliable projects which use QCBMs. 
	
- Implemented a set of core and auxiliary functions.
	
- Run various experiments and tests using simulators, e.g. a Sum of Three Dice (Discrete Gaussian) Experiment and a Bars \& Stripes (BAS) Experiment.
	
- Made first versions of the Jupyter notebooks accompanying the project.  
	
- Structured some of the tools within a documented Python package. 
	

As an example of the results obtained we have the following comparative understanding of the generated distributions in the two examples mentioned above: In the Sum of Three Dice (Discrete Gaussian) Experiment there is an almost perfect matching between the generated distribution and the original distribution, showing high fidelity and minimal information loss in the generation process. On the other hand, in the  Bars \& Stripes (BAS) Experiment there is only a partial match of the generated distribution with the target BAS distribution, fidelity is lower, and precision remains imperfect. 


### A word on Qiskit

The circuit part of a QCBM consists of two kinds of layers:
-  Rotation layers.
- Entanglement layers. 

This structure can be realized as an instance of  the [efficient su2](https://quantum.cloud.ibm.com/docs/en/api/qiskit/qiskit.circuit.library.EfficientSU2) circuit in Qiskit.  Because of the above many of the choices made in the project reflect those underlying this type of circuit as the core component. 


The training loop of the model consist of a series of passes where samples from states prepared by the circuit are taken. Because of this the [Sampler](https://quantum.cloud.ibm.com/docs/en/api/qiskit-ibm-runtime/sampler-v2) primitive plays a central role in the project. 


### Astrolabe

Three works have been specially relevant in guiding the route taken by the project:

- The seminal work of Liu and Wang [LiuWang2018](https://journals.aps.org/pra/abstract/10.1103/PhysRevA.98.062324) where QCBMs are studied in depth.
- The work of Hamilton et. al. [Hamilton2019](https://journals.aps.org/pra/abstract/10.1103/PhysRevA.99.062323) which building upon the above work uses QCBMs to benchmark quantum computers. 
- The work of Ghazi Vakili et. al. [GhaziVakili2025](https://www.nature.com/articles/s41587-024-02526-3) where new inhibitors for KRAS are discovered using a hybrid algorithm with a QCBM as its main quantum component. 

From the first two we have taken our base-test examples for distributions to be learned by the models: Gaussian distributions and the distribution underlying the Bars and Stripes dataset. The Gaussian distribution is also approximated by sum-of-dice simulator. 

The last reference was one of our initial motivations and contains a pipeline for drug discovery that we aim to improve.   
	 
## Challenges

We are currently working towards a pipeline for testing our tools in real quantum hardware. Optimizing this for the resources that are available to us is for now the biggest challenge facing the project.  

## The path forward

Our work in the coming months will focus on four core aspects: 

- Run experiments and tests in quantum hardware.
-  Showcase the tools by implementing a model similar to the one used in \cite{GhaziVakili2025}.
-  Structure the tools developed as a Qiskit addon. 
-  Write all the relevant documentation.
